### PR TITLE
Feature/incorrect error code for api key epmdj 10761

### PR DESCRIPTION
**Pull request by bodyangug**

### CODE REVIEW

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt [hunk 1]
```diff
@@ -41,4 +43,28 @@ class ApiKeyBuilderTest {
...
     }
 
+    @Test
+    fun `given api key with invalid format, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test") }
+    }
+
+    @Test
+    fun `given api key with the first part, parse must throw NotAuthenticatedException`() {
```
Comment (iryabov) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: It's better to say what's missing:
`given api key without secret part, ...`

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt [hunk 2]
```diff
@@ -41,4 +43,28 @@ class ApiKeyBuilderTest {
...
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test_") }
+    }
+
+    @Test
+    fun `given api key with the last part, parse must throw NotAuthenticatedException`() {
```
Comment (iryabov) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: It's better to say what's missing:
`given api key without id part, ...`

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt [hunk 3]
```diff
@@ -41,4 +43,28 @@ class ApiKeyBuilderTest {
...
+    @Test
+    fun `given api key with invalid format, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test") }
+    }
+
+    @Test
+    fun `given api key with the first part, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("test_") }
```
Comment (iryabov) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: `parse("id_")`

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt [hunk 4]
```diff
@@ -41,4 +43,28 @@ class ApiKeyBuilderTest {
...
+    }
+
+    @Test
+    fun `given api key with the last part, parse must throw NotAuthenticatedException`() {
+        val apiKeyBuilder = ApiKeyBuilderImpl()
+        assertThrows<NotAuthenticatedException> { apiKeyBuilder.parse("_test") }
```
Comment (iryabov) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: parse("_secret")

Comment (bodyangug) [reply admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: changed

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt (hunk 3 above)
Comment (bodyangug) [reply admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: changed

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt (hunk 2 above)
Comment (bodyangug) [reply admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: renamed

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt (hunk 1 above)
Comment (bodyangug) [reply admin-auth/src/test/kotlin/com/epam/drill/admin/auth/ApiKeyBuilderTest.kt]: renamed

**Pull request merged**